libs/api.lua

```diff
@@ -130,6 +131,44 @@ end
 
 local metaCache = {}
 
+-- Collect statistics about this server's running instance.
+-- Return these metrics in a table.  The intended consumer of this
+-- information should render it in JSON format for transmission to
+-- a time-series database and/or visualization tools.
+local function collectStats()
+  local entry, fdsUsed = 0, 0
+  local memoryUsed = 1024 * collectgarbage("count")
+
+  local function countFDs(path)
+    local entries = 0
+
+    -- You might think this leaks a filehandle, but it actually doesn't.
+    -- You can confirm this yourself by hitting /metrics several times in a row,
+    -- and observing that lua.fds.total does not increase with each GET request.
+
+    for entry in io.popen([[ls -1 "]]..path..[["]]):lines() do
```

Comment:
Since the rest of lit is based on coroutines, let's use coro-fs here. For in a sample program you would do:

```lua
local fs = require('coro-fs')

-- Create and run coroutine so that we can use coroutine-based APIs
-- This function will only block till the first time the coroutine suspends itself.
coroutine.create(function ()
  for entry in fs.scandir do
    print(entry.name, entry.type)
  end
end)()

```